stdlib/sys.mli

```diff
@@ -354,6 +354,22 @@ val runtime_warnings_enabled: unit -> bool
 
     @since 4.03.0 *)
 
+val enable_float_array_alloc_warning: bool -> unit
+(** Control whether the OCaml runtime system can emit warnings when float
+    arrays are allocated.
```

Comment:
Maybe it is better to be explicit and write `values of type [float array]` to differentiate them from `floatarray` values.

Reply:
Maybe we should be cautious about this. The example Gabriel provided emits the warning even though it is "not" a `float array` on the surface.

Oh well, I guess the main concern is to distinguish from `floatarray`